### PR TITLE
Improve Group, add Group::name, Group::host

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ?
| Fixed issues  | -

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
@rustamwin would you please merge master into the PR branch so tests pass?

File: src/RouteCollection.php [hunk 1]
```diff
@@ -127,6 +127,12 @@ private function injectGroup(Group $group, array &$tree, string $prefix = ''): v
 
             /** @var Route $modifiedItem */
             $modifiedItem = $item->pattern($prefix . $item->getPattern());
+            if ($group->getHost() !== null && $modifiedItem->getHost() === null) {
+                $modifiedItem = $modifiedItem->host($group->getHost());
+            }
+            if ($group->getName() !== null && strpos($modifiedItem->getName(), implode(', ', $modifiedItem->getMethods())) === false) {
```
Comment (vjik) [inline src/RouteCollection.php:133]: What is the mean of checking `strpos($modifiedItem->getName(), implode(', ', $modifiedItem->getMethods())) === false` ?

Comment (rustamwin) [reply src/RouteCollection.php:133]: Checking that the route has a default name. We skip unnamed routes.

Comment (vjik) [reply src/RouteCollection.php:133]: May be create method `$route->isDefaultName(): bool` ?

Comment (rustamwin) [reply src/RouteCollection.php:133]: > May be create method $route->isDefaultName(): bool ?

What do you guys think @yiisoft/yii3?

File: src/GroupParametersInterface.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Router;
+
+interface GroupParametersInterface
```
Comment (vjik) [inline src/GroupParametersInterface.php:7]: Why? Has use case for read groups in application?

Comment (rustamwin) [reply src/GroupParametersInterface.php]: Deleted.

File: src/RouteCollection.php (hunk 1 above)
Comment (samdark) [reply src/RouteCollection.php]: Seems to be alright.

**Review by samdark — APPROVED** [review]

**Pull request merged**